### PR TITLE
Lulu keyboard by boardsource

### PR DESCRIPTION
**Pull request by daysgobye**
## Description
This is the keyboard default map and via map for the lulu keyboard by boardsource.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardsource/lulu/lulu.c [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    oled_write_raw_P(logo, sizeof(logo));
+}
+enum Layers{
+    _QWERTY,
+    _RAISE,
+    _LOWER,
+    _ADJUST
+};
+
+void process_layer_state(void) {
```
Comment (drashna) [inline keyboards/boardsource/lulu/lulu.c:262]: This is neat!

File: keyboards/boardsource/lulu/rules.mk [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE		= yes      # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/boardsource/lulu/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/boardsource/lulu/rules.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+MOUSEKEY_ENABLE			= yes      # Mouse keys
+EXTRAKEY_ENABLE			= yes      # Audio control and System control
+CONSOLE_ENABLE			= no       # Console for debug
+COMMAND_ENABLE			= no       # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE		= no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/boardsource/lulu/rules.mk:17]:
```suggestion
```

File: keyboards/boardsource/lulu/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+NKRO_ENABLE				= yes      # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/boardsource/lulu/rules.mk:18]:
```suggestion
NKRO_ENABLE				= yes      # Enable N-key Rollover
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/boardsource/lulu/lulu.c (hunk 1 above)
Comment (daysgobye) [reply keyboards/boardsource/lulu/lulu.c:262]: thanks lol did you some how see what it looks like? Oh you can look at pics on the site :)

**Review by drashna — APPROVED** [review]

File: keyboards/boardsource/lulu/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+* Hardware Availability: this keyboard is available from the [Boardsource store](https://boardsource.xyz/store/61d0b772319a1f3cc53ba2fb)
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make boardsource/lulu:default
+
+Flashing example for this keyboard:
+
+    make boardsource/lulu:default:dfu
```
Comment (fauxpark) [inline keyboards/boardsource/lulu/readme.md:19]:
```suggestion
    make boardsource/lulu:default:flash
```

File: keyboards/boardsource/lulu/readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Lulu
+
+![boardsource Lulu](https://i.imgur.com/tjvUoXT.png)
```
Comment (fauxpark) [inline keyboards/boardsource/lulu/readme.md:3]:
```suggestion
![boardsource Lulu](https://i.imgur.com/tjvUoXTh.png)
```

File: keyboards/boardsource/lulu/lulu.c [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (fauxpark) [inline keyboards/boardsource/lulu/lulu.c:56]:
```suggestion
```

File: keyboards/boardsource/lulu/rules.mk [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+BACKLIGHT_ENABLE		= no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE			= no       # Enable keyboard RGB underglow
+SPLIT_KEYBOARD			= yes
+RGB_MATRIX_ENABLE		= yes
+RGB_MATRIX_DRIVER		= WS2812
+RGB_MATRIX_SUPPORTED 	= yes
+OLED_ENABLE     		= yes
+OLED_DRIVER     		= SSD1306
+LTO_ENABLE      		= yes
+ENCODER_ENABLE 			= yes
```
Comment (fauxpark) [inline keyboards/boardsource/lulu/rules.mk:25]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
SPLIT_KEYBOARD = yes
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
RGB_MATRIX_SUPPORTED = yes
OLED_ENABLE = yes
OLED_DRIVER = SSD1306
LTO_ENABLE = yes
ENCODER_ENABLE = yes  
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/boardsource/lulu/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *                   | LAlt | LGUI |LOWER | /Space  /       \Enter \  |RAISE |BackSP| RGUI |
+ *                   |      |      |      |/       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------'
+ */
+
+ [_QWERTY] = LAYOUT(
+  KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_GRV,
+  KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS,
+  KC_LCTRL, KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
```
Comment (fauxpark) [inline keyboards/boardsource/lulu/keymaps/default/keymap.c:36]:
```suggestion
  KC_LCTL, KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**